sdk/database/newdbplugin/testing/test_helpers.go

```diff
@@ -0,0 +1,71 @@
+package dbtesting
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/database/newdbplugin"
+)
+
+func AssertInitialize(t *testing.T, db newdbplugin.Database, req newdbplugin.InitializeRequest) newdbplugin.InitializeResponse {
+	t.Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
Can we make this a tunable value? It may be sufficient for certain cases such as mocked databases, but 2 seconds might not be enough if operations end up taking longer than this duration  for certain databases which can happen on CI (depending on load, machine type/size, etc.) and result in flaky tests.

Reply:
I'm hesitant to add too much configurability to these as these are meant for the common use case. If they don't work for a specific database, those tests can fall back to calling the database functions directly.

Reply:
I added the ability to change the timeouts via env variable with a default of 2s.